src/pages/SupplierRequisitionsPage.js

```diff
@@ -38,36 +40,44 @@ export class SupplierRequisitionsPage extends React.Component {
     };
   }
 
-  onDeleteConfirm = () => {
+  componentDidMount() {
+    const { settings, database } = this.props;
+    this.setState({ usesPrograms: !!getAllPrograms(settings, database) });
```

Comment:
Fixed 13cc7a4dc368832eaa0dc68d184f9235a1e02f08

Reply:
@joshxg 
sorry i didn't add a suggestions, getAllPrograms should return an array or null IMO.
This def looks a bit out of the ordinary `!!getAllPrograms(settings, database).length`
 
Thinking more deeply about this, we may we should add a 'coding convention'
* if a method, may return a 'fail' it should be 'null'
or
* method should always return type in header comment docs

Second one sounds like the way to go, but i don't fully agree with it. i.e.
* integer returning method, that fails to find and integer, we would have to use special integer (-1 or something ) to know that method failed, null as generic would be preferred

This is up for discussion but def need a change from `!!getAllPrograms(settings, database).length`

Reply:
done : b81089e992be835526c01c2def469ae87324bd89 (I think?)

---

I'm not very consistent with return values, I need to be. 

Both sides of the argument have merit - I think in JS, it could be better to return null as it is somewhat easier to deal with than in other languages, but not sure.

The argument against your integer example is `Array.prototype.findIndex()`, and other various `findIndex`  methods in almost every language which returns `-1`.

These are not directly related being an OOP patterns: [Null-object pattern](https://en.wikipedia.org/wiki/Null_object_pattern) and [polymorphism over conditionals](https://refactoring.guru/replace-conditional-with-polymorphism) but somewhat relevant I think. Conclusions from these seem to argue for returning an object of the same type. I think quite different for JS though!

Reply:
> The argument against your integer example is Array.prototype.findIndex(), and other various findIndex methods in almost every language which returns -1 (Although one counter-example isn't great. Also I have always found it strange that that this returns -1).

I was never a fan of -1 returns btw.

> These are not directly related being an OOP patterns: Null-object pattern and polymorphism over conditionals but somewhat relevant I think. Conclusions from these seem to argue for returning an object of the same type. I think quite different for JS though!

Would be good to have a group discussion about thi

> done : b81089e (I think?)

looks good